common/containers/Tabs/SendTransaction/components/UnitDropdown.jsx

```diff
@@ -1,87 +1,40 @@
 // @flow
 import React from 'react';
+import SimpleDropDown from 'components/ui/SimpleDropdown';
 
-class Option extends React.Component {
-  props: {
-    value: string,
-    active: boolean,
-    onChange: (value: string) => void
-  };
-  render() {
-    const { value, active } = this.props;
-    return (
-      <li>
-        <a className={active ? 'active' : ''} onClick={this.onChange}>
-          {value}
-        </a>
-      </li>
-    );
-  }
-
-  onChange = () => {
-    this.props.onChange(this.props.value);
-  };
-}
+type UnitDropdownProps = {
+  value: string,
+  options: string[],
+  onChange?: (value: string) => void
+};
 
 export default class UnitDropdown extends React.Component {
-  props: {
-    value: string,
-    options: string[],
-    onChange?: (value: string) => void
-  };
+  props: UnitDropdownProps;
+
   state: {
     expanded: boolean
   } = {
     expanded: false
   };
 
   render() {
-    const { value, options, onChange } = this.props;
-    const isReadonly = !onChange;
+    const { value, options } = this.props;
 
     return (
       <div className="input-group-btn">
-        <a
-          style={{ minWidth: 170 }}
+        <SimpleDropDown
           className="btn btn-default dropdown-toggle"
-          onClick={this.onToggleExpand}
-        >
-          <strong>
-            {value}
-            <i className="caret" />
-          </strong>
-        </a>
-        {this.state.expanded &&
-          !isReadonly &&
-          <ul className="dropdown-menu dropdown-menu-right">
-            {options.map(o =>
-              <Option
-                key={o}
-                active={value === o}
-                value={o}
-                onChange={this.onChange}
-              />
-            )}
-          </ul>}
+          value={value}
+          onChange={this.onChange}
+          options={options}
+        />
       </div>
     );
   }
 
-  onToggleExpand = () => {
-    this.setState(state => {
-      return {
-        expanded: !state.expanded
-      };
-    });
-  };
-
-  onChange = (value: string) => {
-    this.setState({
-      expanded: false
-    });
-
+  onChange = (e: SyntheticInputEvent) => {
     if (this.props.onChange) {
-      this.props.onChange(value);
+      this.props.onChange(e.target.value);
```

Comment:
Perhaps we should keep this interface the same (Just passing back value) rather than having this read in the event. In case we don't end up doing a select list, but instead make it a bootstrap dropdown.

Reply:
Would that really happen at the `UnitDropdown` level though, or wouldn't that be a different component.

I guess we need to define what `UnitDropdown` is and what `SimpleDropDown` is, and how many other dropdowns we need.

Reply:
I would expect it to happen in SimpleDropDown, so that it calls `this.props.onChange(ev.target.value)`